mindmeld/models/containers.py

```
@@ -0,0 +1,374 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import zipfile
+from urllib.request import urlretrieve
+
+import numpy as np
+from tqdm import tqdm
+
+from ._util import _is_module_available, _get_module_or_attr as _getattr
+from ..core import Bunch
+from ..exceptions import EmbeddingDownloadError
+from ..path import (
+    EMBEDDINGS_FILE_PATH,
+    EMBEDDINGS_FOLDER_PATH,
+)
+from ..resource_loader import Hasher
+
+logger = logging.getLogger(__name__)
+
+
+class TqdmUpTo(tqdm):
+    """Provides `update_to(n)` which uses `tqdm.update(delta_n)`."""
+
+    def update_to(self, b=1, bsize=1, tsize=None):
+        """Reports update statistics on the download progress.
+
+        Args:
+            b (int): Number of blocks transferred so far [default: 1].
+            bsize (int): Size of each block (in tqdm units) [default: 1].
+            tsize (int): Total size (in tqdm units). If [default: None] remains unchanged.
+        """
+        if tsize is not None:
+            self.total = tsize
+        self.update(b * bsize - self.n)  # will also set self.n = b * bsize
+
+
+class GloVeEmbeddingsContainer:
+    """This class is responsible for the downloading, extraction and storing of
+    word embeddings based on the GloVe format.
+
+    To facilitate not loading the large glove embedding file to memory everytime a new container is
+    created, a class-level attribute with a hashmap is created.
+    """
+    CONTAINER_LOOKUP = {}
+
+    GLOVE_DOWNLOAD_LINK = "http://nlp.stanford.edu/data/glove.6B.zip"
+    EMBEDDING_FILE_PATH_TEMPLATE = "glove.6B.{}d.txt"
+    ALLOWED_WORD_EMBEDDING_DIMENSIONS = [50, 100, 200, 300]
+
+    def __init__(self, token_dimension=300, token_pretrained_embedding_filepath=None):
+        # validations
+        if token_dimension not in GloVeEmbeddingsContainer.ALLOWED_WORD_EMBEDDING_DIMENSIONS:
```

I think we should fail explicitly here instead of downloading a default size.